book/src/using_rusty/build_description_files.md

```diff
@@ -0,0 +1,122 @@
+
+`rustyc build [Config]`
+
+Note that if `rustyc` cannot find the `plc.json` file, it will throw an error and request the path. The default location for the build file is the current directory. The command for building with an additional path looks like this:
+
+`rustyc build src/plc.json`
+
+
+# Plc.json
+
+For the build description file to work, the build description file must be the [json](https://en.wikipedia.org/wiki/JavaScript_Object_Notation) format. All the keys used in the build description file are described in the following sections. 
+
+
+## files
+
+The key `files` is the equivalent to the `input` parameter, which adds all the `.st` files that needs to be compiled. The value of `files` is an array of strings, definied as followed:
+```json
+"files" : [
+    "examples/hello_world.st",
+    "examples/hw.st"
+]
+```
+
+
+## optimization
+
+`rustyc` offers 4 levels of optimization which correspond to the levels established by llvm respectively [clang](https://clang.llvm.org/docs/CommandGuide/clang.html#code-generation-options) (`none` to `agressive`).
+To use an optimization, the key `optimization` is required:
+- `"optimization" : "none"`
+- `"optimization" : "less"`
+- `"optimization" : "default"`
+- `"optimization" : "aggressive"`
+
+By default `rustyc` will use `default` which corresponds to clang's `-o2`.
+
+
+## error_format
+
+`rustyc` offers 2 levels of formatting errors. To specify which error format is wanted, the key `error_format` is required:
+- `"error_format" : "Clang"`   This is used to get fewer error messaged
+- `"error_format" : "Rich"`    This is used to get a verbose error description. 
+
+
+## libraries
+
+To link several executables `rustyc` has the option to add libraries and automatically build and like them together. if no compile type has been selected `rustyc` will link the files on default.
+
+```json
+"libraries" : [
+    {
+        "name" : "iec61131std",
+        "path" : "path/to/lib/",
+        "include_path" : [
+            "examples/hw.st",
+            "examples/hello_world.st"
+        ]
+    }
+]
+```
+
+## output
+
+Similarly to specifying an output file via the `-o` or `--output` option, in the build file we use `"output" : "output.so"` to define the output file. The default location is likewise to the location for the build file, namely the current directory. 
+
+
+
+## Optional Keys
+### sysroot
+
+`rustyc` is using the `sysroot` key for linking purposes. It is considered to be the root directory for the purpose of locating headers and libraries.
+
+
+### target
+
+To build and compile [structured text](https://en.wikipedia.org/wiki/Structured_text) for the rigth platform we need to specify the `target`. As `rustyc` is using [LLVM](https://en.wikipedia.org/wiki/LLVM) a target-tripple supported by LLVM needs to be selected. The default `target` is `x86_64-linux-gnu`.
```

Comment:
I just noticed this, the default target is the host machine's target.
So if you are using the dev container on an x86_64 docker it is x86_64-linux-gnu